### PR TITLE
GitHub Issue NOAA-EMC/emcpy#95: Add spectrum statistics capability

### PR DESCRIPTION
**Pull request by ADCollard**
This pull request is related to #95 

I am adding @aerorahul as I am not sure if this is too close to code developed in PyGSI.  Adding @EdwardSafford-NOAA as I am not sure how this fits in with Radiance Monitor re-engineering.

### CODE REVIEW

File: src/emcpy/stats.py [hunk 1]
```diff
@@ -285,3 +286,45 @@ def bootstrap(insample, level=.95, estimator='mean', nrepl=10000):
     ci_upper = _np.percentile(deltas, upper_pctile)
 
     return ci_lower, ci_upper
+
+
+def spectrum_stats(df):
+    '''
```
Comment (EdwardSafford-NOAA) [inline src/emcpy/stats.py:292]: Do single quotes work to generate the automatic documentation?  I don't know the answer but I think, heretofore, double quotes have been used.  Maybe @CoryMartin-NOAA  can speak to that.

File: src/emcpy/stats.py [hunk 2]
```diff
@@ -285,3 +286,45 @@ def bootstrap(insample, level=.95, estimator='mean', nrepl=10000):
...
+
+    # Loop through channels and calculate stats
+    for chan in sc:
+        # Subset based on channel and QC indexes
+        tmp = dfs.loc[(chan,0.0)]
+        # This next line makes sure that all rejected obs are avoided.
+        # Using the inverse observation error is more reliable than the QC flag.
+        tmp = tmp[(tmp['inverse_observation_error'] > 0.0)]
+        count = tmp.size
+        channel_stats["count"].loc[chan] = count
```
Comment (EdwardSafford-NOAA) [inline src/emcpy/stats.py]: Not a big deal but you could skip the assignment to count and just set channel_stats["count"].loc[chan] = tmp.size.

**Review by EdwardSafford-NOAA — COMMENTED** [review]
Looks good to me.  I have some comments but no real problems.

File: src/emcpy/stats.py (hunk 2 above)
Comment (ADCollard) [reply src/emcpy/stats.py]: @EdwardSafford-NOAA You are right.  I have made the change.

**Comment by CoryMartin-NOAA** [comment]
@ADCollard I think this probably is a better fit for PyGSI, as it looks for things like `inverse_observation_error`

**Comment by ADCollard** [comment]
> 

@CoryMartin-NOAA You are probably right that this could and should go in PyGSI.  I'm not sure exactly where though.  Should I start a new file (`stats.py`?)for stats utilities over there?

**Comment by CoryMartin-NOAA** [comment]
@ADCollard yeah I think that makes sense to me.

**Comment by ADCollard** [comment]
Closing pull request as I have opened a new one in [PyGSI](https://github.com/NOAA-EMC/PyGSI/pull/117).

**Pull request closed without merging**